src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1108,6 +1125,282 @@ bool os::dll_address_to_library_name(address addr, char* buf,
   return true;
 }
 
+// get the library search path burned in to the executable file during linking
+// If the libpath cannot be retrieved return an empty path
+static const char* rtv_linkedin_libpath() {
+  static char buffer[4096];
+  static const char* libpath = 0;
+
+  if (libpath)
+    return libpath;
+
+  char pgmpath[32+1];
+  // retrieve the path to the currently running executable binary
+  snprintf(pgmpath, 32, "/proc/%ld/object/a.out", (long)getpid());
+  // open the currently running executable binary
+  FILE *f = fopen(pgmpath, "r");
+  // read the generic XCOFF header and analyze the substructures
+  // to find the burned in libpath
+  fread(buffer, 1, FILHSZ_64 + _AOUTHSZ_EXEC_64, f);
+
+  if (((struct filehdr*)buffer)->f_magic == U802TOCMAGIC ) {
+    // __XCOFF32__
+    struct xcoffhdr xcoff;
+    struct scnhdr scn;
+    struct ldhdr ldr;
+    memcpy((char*)&xcoff, buffer, FILHSZ + _AOUTHSZ_EXEC);
+    int ldroffset = FILHSZ + xcoff.filehdr.f_opthdr + (xcoff.aouthdr.o_snloader -1)*SCNHSZ;
+    fseek (f, ldroffset, SEEK_SET);
+    fread(buffer, 1, SCNHSZ, f);
+    memcpy((char*)&scn, buffer, SCNHSZ);
+    if (strcmp(scn.s_name, ".loader")) {
+      buffer[0] = 0;
+      libpath = buffer;
+      fclose(f);
+      return libpath;
+    }
+    fseek (f, scn.s_scnptr, SEEK_SET);
+    fread(buffer, 1, LDHDRSZ, f);
+    memcpy((char*)&ldr, buffer, LDHDRSZ);
+    fseek (f, scn.s_scnptr + ldr.l_impoff, SEEK_SET);
+    fread(buffer, 1, 4096, f);
+  }
+  else if (((struct filehdr*)buffer)->f_magic == U64_TOCMAGIC ) {
+    // __XCOFF64__
+    struct _S_(xcoffhdr) xcoff64;
+    struct _S_(scnhdr) scn64;
+    struct _S_(ldhdr) ldr64;
+    memcpy((char*)&xcoff64, buffer, FILHSZ_64 + _AOUTHSZ_EXEC_64);
+    int ldroffset = FILHSZ_64 + xcoff64.filehdr.f_opthdr + (xcoff64.aouthdr.o_snloader -1)*SCNHSZ_64;
+    fseek (f, ldroffset, SEEK_SET);
+    fread(buffer, 1, SCNHSZ_64, f);
+    memcpy((char*)&scn64, buffer, SCNHSZ_64);
+    if (strcmp(scn64.s_name, ".loader")) {
+      buffer[0] = 0;
+      libpath = buffer;
+      fclose(f);
+      return libpath;
+    }
+    fseek (f, scn64.s_scnptr, SEEK_SET);
+    fread(buffer, 1, LDHDRSZ_64, f);
+    memcpy((char*)&ldr64, buffer, LDHDRSZ_64);
+    fseek (f, scn64.s_scnptr + ldr64.l_impoff, SEEK_SET);
```

Comment:
nit: please use consistent spacing according to hotspot rules. here, remove space.

Reply:
Do you mean the space `fseek (` ? 
Done.